### ISSUE TITLE
Implement normalization layers

### ISSUE THREAD

**Issue by francescofarina** [mlx-graphs/mlx-graphs#116]
Common graph-specific normalization layers are currently missing and in general very useful to have.

Examples would be `InstanceNorm`, `LayerNorm`, `GraphNorm` (see, e.g. http://proceedings.mlr.press/v139/cai21e/cai21e.pdf)

**Comment by thegodone** [comment]
I think most important are LayerNorm and BatchNorm to begin with

**Comment by TristanBilot** [comment]
It may be easier to follow PyG's implementation: [LayerNorm](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/norm/layer_norm.html#LayerNorm), [GraphNorm](https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/norm/graph_norm.html#GraphNorm).
